tests/integration/test_charm.py

```diff
@@ -1,57 +1,54 @@
 #!/usr/bin/env python3
-# Copyright 2022 Canonical Ltd.
+# Copyright 2024 user
 # See LICENSE file for licensing details.
 
-import asyncio
 import logging
+import shlex
 from pathlib import Path
 
-import yaml
 import pytest
-import shlex
+import yaml
 from pytest_operator.plugin import OpsTest
 
-log = logging.getLogger(__name__)
+logger = logging.getLogger(__name__)
 
-METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+METADATA = yaml.safe_load(Path("./charmcraft.yaml").read_text())
 APP_NAME = METADATA["name"]
-READY_MESSAGE = "Ready to test."
 TEST_ACTION_NAME = "test"
-# Allow e2e tests to run with up to 2 non-ready nodes 
+READY_MESSAGE = "Ready to test."
+# Allow e2e tests to run with up to 2 non-ready nodes
 EXTRA_ARGS = "-allowed-not-ready-nodes 2"
 # Run reduced test suite
 SKIP_TESTS = r"\[(Flaky|Slow|Conformance|Feature:.*)\]"
 
 
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test: OpsTest):
-    """Build the kubernetes-e2e charm and deploy it 
-    
+    """Build the kubernetes-e2e charm and deploy it.
+
     Assert on the unit status before any relations/configurations take place.
     """
-    log.info("Build charm...")
+    logger.info("Build charm...")
     charm = await ops_test.build_charm(".")
 
     overlays = [
-        ops_test.Bundle("charmed-kubernetes", channel="1.29/stable"),
+        ops_test.Bundle("charmed-kubernetes", channel="1.29/beta"),
```

Comment:
Could we use the `kubernetes-core` bundle instead?

Reply:
I've been using `juju deploy kubernetes-core` this whole time, would it be the equivalent ?

Reply:
i think the e2e needs at least  2 workers... `charmed-kubernetes` might be right here

Reply:
but definitely use the `edge` channel here  (yes, `edge` should work)
```suggestion
        ops_test.Bundle("charmed-kubernetes", channel="edge"),
```